src/reactjs/tests/ssr-hydration.test.tsx

```diff
@@ -79,14 +115,17 @@ describe('Server side rendering with de/rehydration', () => {
 
     // Check that we have no React hydration mismatches
     expect(mockLogger.error).not.toHaveBeenCalled()
-    expect(fetchDataSuccess).toHaveBeenCalledTimes(1)
+    expect(fetchDataSuccess).toHaveBeenCalledTimes(2)
```

Comment:
The point of this expect is to check that we did not have to actually go and fetch the data on the client, because hydration worked and it was already there. Was this a temporary fix to get tests working for now, or was there some other reason?

Reply:
I dug into this, and I think something was fishy with the tests to begin with. As we've documented, since `staleTime` defaults to 0, unless you set it to something else we automatically refetch on mount. So this should probably have been 2 all along and these new tests and wrapping the new hydration in an `act` uncovered that.

I think we should update these tests to have a `staleTime` instead and change this back to 1 (requires updating expectedMarkup as well). Then we can also run them in both the 17 and 18 variations. I can take care of that while working on hydration.

Reply:
@Ephem I think this is the same conclusion I cam to when changing it:

https://github.com/tannerlinsley/react-query/commit/a4b456431519dae67f6005602b81f9eedbb6b302

Changing the test to make sure we don't fetch on the client by setting a staleTime is a good idea 👍

Reply:
I read the linked discussion, sorry for causing that confusion in the first place.. 😄  I think it would have made sense to write the tests with a set staleTime to begin with (the common case for this) and have a single test to test the refetch with staleTime 0.